### ISSUE TITLE
urllib error persistently occurs

### ISSUE THREAD

**Issue by joshelbahrawy** [ranaroussi/yfinance#586]
After installing yfinance, everything works fine for a couple of executions of my script. Inevitably, however, I will get this error:
```x =yf.Ticker("MSFT")
>>> x.info
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/josh/.local/lib/python3.8/site-packages/yfinance/ticker.py", line 138, in info
    return self.get_info()
  File "/home/josh/.local/lib/python3.8/site-packages/yfinance/base.py", line 446, in get_info
    self._get_fundamentals(proxy)
  File "/home/josh/.local/lib/python3.8/site-packages/yfinance/base.py", line 285, in _get_fundamentals
    holders = _pd.read_html(url)
  File "/home/josh/.local/lib/python3.8/site-packages/pandas/util/_decorators.py", line 296, in wrapper
    return func(*args, **kwargs)
  File "/home/josh/.local/lib/python3.8/site-packages/pandas/io/html.py", line 1086, in read_html
    return _parse(
  File "/home/josh/.local/lib/python3.8/site-packages/pandas/io/html.py", line 898, in _parse
    tables = p.parse_tables()
  File "/home/josh/.local/lib/python3.8/site-packages/pandas/io/html.py", line 217, in parse_tables
    tables = self._parse_tables(self._build_doc(), self.match, self.attrs)
  File "/home/josh/.local/lib/python3.8/site-packages/pandas/io/html.py", line 736, in _build_doc
    raise e
  File "/home/josh/.local/lib/python3.8/site-packages/pandas/io/html.py", line 717, in _build_doc
    with urlopen(self.io) as f:
  File "/home/josh/.local/lib/python3.8/site-packages/pandas/io/common.py", line 137, in urlopen
    return urllib.request.urlopen(*args, **kwargs)
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

If I pip uninstall yfinance and re-install it, it works fine again for a little while again and then starts failing again.

**Comment by yacoubb** [comment]
I get the same error, but reinstalling doesn't seem to fix it for me.

**Comment by yacoubb** [comment]
Seems like `ticker.get_info()` fails, while `ticker.history()` for yesterday's prices works fine.

**Comment by ehmadzubair** [comment]
Any fix for this error?

**Comment by ProgrammerInvestor** [comment]
Have anybody fixed it?

**Comment by Hoggie-92** [comment]
Anybody fixed this one yet? For me yfinance was working fine up until today (even yesterday it was all good), now I am persistently seeing 404 errors. At first glance it appears to be related to pandas read_html function. It looks like this function has only been used the once on line 291 of the base.py file.

**Comment by Hoggie-92** [comment]
Hey everyone, for anyone dealing with this issue, I managed to get it to work by updating the base.py file (my version of yfinance is older 0.1.54 and it looks like this may even have been fixed in the most recent version 0.1.59).
Replace the following:

 holders = _pd.read_html(url)

With:

r = _requests.get(url,headers={'User-Agent': 'Mozilla/5.0'})
holders = _pd.read_html(r.content)

**Comment by cardiganman** [comment]
I had this problem starting yesterday.  I checked my yfinance version and it was 1.54.  Updating yfinance to version 1.6 solved the problem.

**Comment by emogee** [comment]
Updating my yfinance also fixed my problem.

However, until recently, I have been a devout conda user.  I was disappointed to find that using anaconda to update yfinance did not cause it to update to a newer version.  

So, instead, I uninstalled yfinance using conda and then I did this (found via google)...

pip install -i https://pypi.anaconda.org/ranaroussi/simple yfinance

...and then problem solved.  (I am not positive if uninstalling yfinance with conda is a required first step).
